### PR TITLE
move Libdvd to depends

### PR DESCRIPTION
**Pull request by wsnipex**
this moves libdvd to depends, pulling from our own repositories. 

I gave it a quick try on linux with a test dvd iso, but it gives an error.

> 19:17:12 T:140592437192448    INFO:   msg: libdvdnav: Using dvdnav version 5.0.4
> 19:17:12 T:140592437192448 WARNING: dll_ioctl - 21392 request failed with error [25] Inappropriate ioctl for device
> 19:17:12 T:140592437192448    INFO:   msg: libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
> 19:17:12 T:140592437192448    INFO:   msg: libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.BUP failed

 No idea if the problem is the (possibly missing) patches in our repo, my changes or something else completely. In any case, this still needs work and testing on all platforms.

@Voyager1 @FernetMenta @afedchin 
@fetzerch fyi, this breaks cmake once again. Unless you beat me to it, I'll take care once this works with autotools.

### CODE REVIEW

**Comment by Voyager1** [comment]
@wsnipex the file open error is also known on Windows. It works fine with real local FS, anything to be accessed through VFS seems broken. See https://github.com/xbmc/xbmc/pull/9120#issuecomment-183948666

File: lib/libdvd/Makefile.in [hunk 1]
```diff
@@ -3,80 +3,89 @@ ARCH=@ARCH@
...
-$(SYSDIR)/libdvdnav-$(ARCH).so: $($(WRAPPER)) $(DVDCSS_A) libdvdread/obj/libdvdread.a libdvdnav/obj/libdvdnav.a
+		$(WRAPPER) libdvdcss/*.o $(BUNDLE1_O)
+
+$(SYSDIR)/libdvdnav-$(ARCH).so: $(WRAPPER) $(DVDNAV_A) $(DVDREAD_A) $(DVDCSS_A)
+	[ -d libdvdread ] || mkdir libdvdread
+	[ -d libdvdnav ] || mkdir libdvdnav
+	[ -d libdvdcss ] || mkdir libdvdcss
+	cd libdvdcss; ar x $(DVDCSS_A)
+	cd libdvdnav; ar x $(DVDNAV_A)
+	cd libdvdread; ar x $(DVDREAD_A)
```
Comment (stefansaraev) [inline lib/libdvd/Makefile.in]: well. after moving to depends, aren't those supposed to be linked like -ldvdcss -ldvdnav -lfoo 
why caling into depends from here and then unpacking the archives ?

**Comment by notspiff** [comment]
doing this while relying on old wrapping hacks is .. suboptimal. do it properly if you want to do it.

**Comment by wsnipex** [comment]
trust me, I wanted to get rid of this crap and just statically link, libs are small after all. 
@FernetMenta wanted to keep it dyloaded and this is the result. Getting the wrapper working for cross and native compile actually took the better part of the day...

I'm not going to rewrite core code I have no knowledge in for a feature I hardly use myself.
Any fixes would be greatly appreciated.

**Comment by FernetMenta** [comment]
@wsnipex thanks! I will have a closer look this evening. Sure, in the long term it makes more sense to get rid of the hacks but for the time being this blocks msys2 upgrade. I think it makes more sense to move forward in iterations, means keeping libdvd as it was (just upgrade). In a later iteration we get drop the wrapper.

**Comment by ace20022** [comment]
@Voyager1 @FernetMenta I want to replace our udf impl with libudfread anyway, so I could look at the callbacks too, but I can't give a date. I just want to avoid that 3 devs work at the same thing.

**Comment by Voyager1** [comment]
@ace20022 - I'd like to get libdvd 5.0.4 working with the old wrapper and msys2 if possible (I think it's compiler/linker settings related). But I'm not going to touch the callbacks thing. So please feel free ...

**Comment by akva2** [comment]
the main problem is that the hook in libdvd is compile time only. it's trivial https://github.com/xbmc/libdvdcss/blob/master/src/device.c#L77 to hook in our io but it cannot be done for precompiled libraries.

**Comment by wsnipex** [comment]
jenkins build this please

**Comment by wsnipex** [comment]
updated. I noticed that tarballs of git branches were never re-downloaded, so updates to the repos were ignored. Fixed by resolving the git hash and using this in the filename. @afedchin you'll probably need something similar for win32.
jenkins build this please

**Comment by wsnipex** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
jenkins build this please

**Comment by afedchin** [comment]
@wsnipex any news on this?

**Comment by wsnipex** [comment]
doesn't build on IOS yet(as usual), cause it lacks ioctls dvdcss needs. That means I have to reintroduce an option to disable it, which I really wanted to avoid.

**Comment by wsnipex** [comment]
jenkins build this please

**Comment by wsnipex** [comment]
jenkins build this please

**Comment by wsnipex** [comment]
sigh, jenkins build this please

**Comment by wsnipex** [comment]
jenkins build this please

**Comment by wsnipex** [comment]
ok, its finally working. I only tested linux though.

**Comment by FernetMenta** [comment]
great! I will test on OSX now

**Comment by wsnipex** [comment]
any test results?

**Comment by FernetMenta** [comment]
looks good on OSX

**Comment by wsnipex** [comment]
I guess we should also test droid and ios, but up to you.
Your button @FernetMenta and @afedchin

**Comment by MartijnKaijser** [comment]
can do some tests tonight on droid

**Comment by MartijnKaijser** [comment]
fails to build on win32 at least

**Comment by wsnipex** [comment]
yes, cause the win32 part is https://github.com/xbmc/xbmc/pull/9120

**Comment by FernetMenta** [comment]
@wsnipex thanks very much!
@afedchin I think it is best if you merge this PR with yours

**Comment by afedchin** [comment]
@FernetMenta oki, will cherry-pick this a little later.

**Comment by MartijnKaijser** [comment]
all fine on android with DVD menus

**Comment by afedchin** [comment]
@FernetMenta all done. #9120 was updated. Jenkins is happy.

**Comment by FernetMenta** [comment]
inject :)

**Comment by afedchin** [comment]
close this in favor of #9120

**Pull request closed without merging**